pkg/artifactory/resource/security/resource_artifactory_distribution_public_key.go

```diff
@@ -0,0 +1,150 @@
+package security
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/jfrog/terraform-provider-shared/packer"
+	"github.com/jfrog/terraform-provider-shared/predicate"
+	"github.com/jfrog/terraform-provider-shared/util"
+)
+
+const DistributionPublicKeysAPIEndPoint = "artifactory/api/security/keys/trusted"
+
+type distributionPublicKeyPayLoad struct {
+	KeyID       string `hcl:"key_id" json:"kid"`
+	Alias       string `hcl:"alias" json:"alias"`
+	Fingerprint string `hcl:"fingerprint" json:"fingerprint"`
+	PublicKey   string `hcl:"public_key" json:"key"`
+	IssuedOn    string `hcl:"issued_on" json:"issued_on"`
+	IssuedBy    string `hcl:"issued_by" json:"issued_by"`
+	ValidUntil  string `hcl:"valid_until" json:"valid_until"`
+}
+
+type DistributionPublicKeysList struct {
+	Keys []distributionPublicKeyPayLoad `json:"keys"`
+}
+
+func ResourceArtifactoryDistributionPublicKey() *schema.Resource {
+
+	type keyPost struct {
+		Alias     string `json:"alias"`
+		PublicKey string `json:"public_key"`
+	}
+
+	var distributionPublicKeySchema = map[string]*schema.Schema{
+		"key_id": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Returns the key id by which this key is referenced in Artifactory.",
+		},
+		"alias": {
+			Type:         schema.TypeString,
+			Required:     true,
+			Description:  "Will be used as a identifier when uploading/retrieving the public key via REST API.",
```

Comment:
Don't hate me 🙈 

'will be used as AN identifier'

Reply:
should be fixed in latest commit